R/pandoc.R

```diff
@@ -749,7 +749,12 @@ pandoc_citeproc <- function() {
 #' @export
 pandoc_lua_filter_args <- function(lua_files) {
   # Lua filters was introduced in pandoc 2.0
-  if (pandoc2.0()) c(rbind("--lua-filter", pandoc_path_arg(lua_files)))
+  if (pandoc2.0()) {
+    if (!nzchar(Sys.getenv("RMARKDOWN_LUA_SHARED"))) {
+      Sys.setenv(RMARKDOWN_LUA_SHARED = pkg_file_lua("shared.lua"))
```

Comment:
When setting options or env vars, usually we need to unset them on exit.

```suggestion
      vars <- xfun::set_envvar(c(RMARKDOWN_LUA_SHARED = pkg_file_lua("shared.lua")))
      on.exit(xfun::set_envvar(vars), add = TRUE)
```

But this needs to be in another higher-level function such as `render()` or `pandoc_convert()`.

Reply:
🤔 I see. 

Does this mean it is better to set this env var elsewhere ? 

I find `pandoc_convert()` to low level because you can use it without any **rmarkdown** specific feature, especially Lua filters. 

Maybe I should set that in `render()` then. 

Otherwise, what about the `on_exit=` argument of our `output_format` when our Lua filters are used ?

Reply:
I am still thinking about the best way to do it. 
I experimented with a few places and I am not yet satisfied. `pandoc_lua_filter_args` maybe not be the right place as it is not tied to the filters that requires this file.

Reply:
I decide that `render()` may be the right place as it concerns setting environment during a rendering of a document, and because this is where lua argument is passed to Pandoc. 

This allows for setting the on.exit() more easily too.